api/api_ethereum.go

```diff
@@ -907,7 +912,11 @@ func (api *EthereumAPI) GetTransactionCount(ctx context.Context, address common.
 func (api *EthereumAPI) GetTransactionByHash(ctx context.Context, hash common.Hash) (*EthRPCTransaction, error) {
 	// Try to return an already finalized transaction
 	if tx, blockHash, blockNumber, index := api.publicTransactionPoolAPI.b.ChainDB().ReadTxAndLookupInfo(hash); tx != nil {
-		return newEthRPCTransaction(tx, blockHash, blockNumber, index), nil
+		block, err := api.publicTransactionPoolAPI.b.BlockByHash(ctx, blockHash)
+		if block == nil || err != nil {
```

Comment:
If transaction is not processed yet, do we return an error? I thought we need to still return without block info.

Reply:
good point, thanks

Reply:
@jimni1222 You mean returning `(nil, nil)`? You are worried about returning info. not related with this API context, right?

Reply:
I mean, even tx is not processed yet(which means not included in the block), still GetTransactionByHash should return with empty block information.
So i think what you can do is, simply remove `block == nil` condition check here.
Then `newEthRPCTransaction` function will handle `block == nil` case ([here](https://github.com/klaytn/klaytn/pull/1452/files/514f419d967617307ebe0f337b854871a994e51d#diff-e5df1f3668663dc2f0acf56e7d1cbdcc01c6f4b349b63e36850597cb5be4f03cL776) and [here](https://github.com/klaytn/klaytn/pull/1452/files/514f419d967617307ebe0f337b854871a994e51d#diff-e5df1f3668663dc2f0acf56e7d1cbdcc01c6f4b349b63e36850597cb5be4f03cR796))

Reply:
Ahh, I got it. Thank you for explanation in detail. 
Then I think we shouldn't return the `error` because `BlockByHash()` only return the `error`, "the block does not exist" when the block not found. We can leave for `newEthRPCTransaction` to process it.

Reply:
Oh, i took a look logic more and i realized i didn't think correctly.

When `ReadTxAndLookupInfo` returns an empty block hash, also returns empty tx object, so `tx != nil` condition should include when transaction is not processed.
If tx is not nil and `ReadTxAndLookupInfo ` returns block hash and number correctly, then `block == nil` should not happen in normal case when `api.publicTransactionPoolAPI.b.BlockByHash(ctx, blockHash)`.

So can you add another error handling logic for that case to handle `BlockByHash` return nil as a block, even we got blockHash and blockNumber from `ReadTxAndLookupInfo` above?